compiler/rustc_ast/src/ast.rs

```diff
@@ -167,10 +167,7 @@ pub enum GenericArgs {
 
 impl GenericArgs {
     pub fn is_angle_bracketed(&self) -> bool {
-        match *self {
-            AngleBracketed(..) => true,
-            _ => false,
-        }
+        matches!(*self, AngleBracketed(..))
```

Comment:
Since you're editing this already, maybe rely on match ergonomics instead of explicitly dereferencing?
```suggestion
        matches!(self, AngleBracketed(..))
```
Looks a tiny bit nicer IMO. Same below for other applicable places.